File: c#/DisgustingEncryption/AES/Aes_128_Ecb_002.cs
namespace DisgustingEncryption
{
    using System;
    using System.Security.Cryptography;
    using System.Text;

    public class Aes_128_Ecb_002
    {
        public static void Show()
        {
            var key1 = "12345678";
            var str1 = "123";

            var enc1 = AesEncrypt(str1, key1);
            // 9FB79E399D9D174358A20FFAB0DE06FB
            Console.WriteLine("after encrypt");
            Console.WriteLine(enc1);

            var dec1 = AesDecrypt(enc1, key1);
            Console.WriteLine("after decrypt");
            Console.WriteLine(dec1);


            var key2 = "1234567812345678";
            var str2 = "你好abcd1234";

            var enc2 = AesEncrypt(str2, key2);
            // 52F86DBA5B6C4F778395BBEDC3E0B489
            Console.WriteLine("after encrypt");
            Console.WriteLine(enc2);

            var dec2 = AesDecrypt(enc2, key2);
            Console.WriteLine("after decrypt");
            Console.WriteLine(dec2);
        }


        public static string AesEncrypt(string toEncrypt, string key)
        {
            key = key.Length > 16 ? key.Substring(0, 15) : key.Length < 16 ? key.PadRight(16, '0') : key;

            byte[] keyArray = Encoding.UTF8.GetBytes(key);
            byte[] toEncryptArray = Encoding.UTF8.GetBytes(toEncrypt);

            RijndaelManaged rDel = new RijndaelManaged
            {
                Key = keyArray,
                Mode = CipherMode.ECB,
                Padding = PaddingMode.PKCS7,
                BlockSize = 128
            };

            ICryptoTransform cTransform = rDel.CreateEncryptor();

            byte[] resultArray = cTransform.TransformFinalBlock(toEncryptArray, 0, toEncryptArray.Length);

            return BytesToHex(resultArray);
        }
  
        public static string AesDecrypt(string toDecrypt, string key)
        {
            key = key.Length > 16 ? key.Substring(0, 15) : key.Length < 16 ? key.PadRight(16, '0') : key;
            byte[] keyArray = Encoding.UTF8.GetBytes(key);            
            byte[] toDecryptArray = HexToBytes(toDecrypt);

            RijndaelManaged rDel = new RijndaelManaged
            {
                Key = keyArray,
                Mode = CipherMode.ECB,
                Padding = PaddingMode.PKCS7,
                BlockSize = 128
            };

            ICryptoTransform cTransform = rDel.CreateDecryptor();

            byte[] resultArray = cTransform.TransformFinalBlock(toDecryptArray, 0, toDecryptArray.Length);

            return Encoding.UTF8.GetString(resultArray);
        }

        private static string BytesToHex(byte[] bytes)
        {
            var hexString = BitConverter.ToString(bytes, 0).Replace("-", string.Empty).ToUpper();
            return hexString;

        }

        private static byte[] HexToBytes(string hexString)
        {
            var bytes = new byte[hexString.Length / 2];
            for (var x = 0; x < bytes.Length; x++)
            {
                var i = Convert.ToInt32(hexString.Substring(x * 2, 2), 16);
                bytes[x] = (byte)i;
            }
            return bytes;
        }

    }
}

File: python/disgustingEncryption/AES/aes_128_ecb_002.py
from Crypto.Cipher import AES
from Crypto.Util.Padding import pad, unpad
import base64
import hashlib

class aes_128_ecb_002:

    def add_to_16(self, value):

        l = len(value)

        if l > 16:
            value = value[:15]
        elif l < 16:
            value =  value + "0" * (16 -l)
         
        return value
  
    def pkcs7padding(self, text):

        bs = AES.block_size 
        print(bs)
        length = len(text)
        bytes_length = len(bytes(text, encoding='utf-8'))

        padding_size = length if(bytes_length == length) else bytes_length
        padding = bs - padding_size % bs
        
        padding_text = chr(padding) * padding
        return text + padding_text


    def pkcs7unpadding(self, text):
        length = len(text)
        unpadding = ord(text[length-1])
        return text[0:length-unpadding]        

    def aes_encrypt(self, content, key):

        key_bytes = bytes(self.add_to_16(key), encoding='utf-8')
       
        aes = AES.new(key_bytes, AES.MODE_ECB)

        content_padding = self.pkcs7padding(content)
        encrypt_bytes = aes.encrypt(bytes(content_padding, encoding='utf-8'))
        
        result = encrypt_bytes.hex().upper()
        return result

    def aes_decrypt(self, content, key):

        key_bytes = bytes(self.add_to_16(key), encoding='utf-8')           

        aes = AES.new(key_bytes, AES.MODE_ECB) 

        encrypt_bytes = bytes.fromhex(content)
        decrypt_bytes = aes.decrypt(encrypt_bytes)  

        result = str(decrypt_bytes, encoding='utf-8')
        result = self.pkcs7unpadding(result)
        
        return result 

    def show(self):
        mystr1 = "123"
        mykey1 = "12345678"

        enc1 = self.aes_encrypt(mystr1, mykey1)
        # 9FB79E399D9D174358A20FFAB0DE06FB
        print("after encrypt")
        print(enc1)        

        dec1 = self.aes_decrypt(enc1, mykey1)
        print("after decrypt")
        print(dec1)        

        mystr2 = "你好abcd1234"
        mykey2 = "1234567812345678"

        
        enc2 = self.aes_encrypt(mystr2, mykey2)        
        # 52F86DBA5B6C4F778395BBEDC3E0B489
        print("after encrypt")
        print(enc2)        

        dec2 = self.aes_decrypt(enc2, mykey2)
        print("after decrypt")
        print(dec2)        


